src/utils/json-data.js

```diff
@@ -49,31 +48,53 @@ const buildGlobalDataFromJson = () => {
       throw new shared.StachePluginError(error.message);
     }
 
-    acc[propertyName] = JSON.parse(contents);
+    buildJsonDataObject(acc, propertyNameSegments, JSON.parse(contents));
 
     return acc;
   }, {});
 
   return dataObject;
 };
 
-const convertFileNameToObjectPropertyName = (fileName) => {
-  return fileName.split('.')[0]
-    .replace(/\s+/g, '_')     // Replace spaces with underscores
-    .replace(/[^\w-]+/g, '')  // Remove all non-word chars
-    .replace(/-/g, '_')       // Replace all dashes with underscores
-    .replace(/--+/g, '_')     // Replace multiple dashes with single underscore
-    .replace(/^-+/, '')       // Trim - from start of text
-    .replace(/-+$/, '');      // Trim - from end of text;
+const convertFileNameToObjectPropertyNameSegments = (fileName) => {
+  fileName =  fileName.split('.')[0]
+    .replace(/\\+/g, '.')      // Replace back-slashes with a single period
+    .replace(/\/+/g, '.')      // Replace slashes with a single period
+    .replace(/\s+/g, '_')      // Replace spaces with underscores
+    .replace(/[^.\w-]+/g, '')  // Remove all non-word chars other than periods
+    .replace(/-/g, '_')        // Replace all dashes with underscores
+    .replace(/--+/g, '_')      // Replace multiple dashes with single underscore
+    .replace(/^-+/, '')        // Trim - from start of text
+    .replace(/-+$/, '');       // Trim - from end of text;
+
+  // Split up nested property name in to an array of valid keys and return
+  return fileName.split('.').filter(key => { return !!key });
```

Comment:
What's an example of a key that would make it through the regex above but equate to false here?

Reply:
If you have an instance where your string is `.directory.file` the array created by the split will be ['', 'directory', 'file'] this ensures that we only ever have ['directory', 'file']